content/3_Silver/Prefix_Sums_2.mdx

```diff
@@ -102,11 +102,94 @@ export const problems = {
 
 <FocusProblem problem={problems.maxsum} />
 
+### Solution
+
+<Spoiler title="Solution">
+
+Consider the prefix sum array. The maximum subarray sum is the maximum possible $p[i]-p[j]$ where $j\leq i$. We can iterate over each index $i$ and keep a running minimum $p[j]$ value. This yields the maximum subarray sum for each possible right bound, and the maximum over all these values is our answer.
+
+<LanguageSection>
+
+<CPPSection>
+
+C++ code:
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+
+int main(){
+	int n; cin >> n;
+	vl dp(n+1, 0); // Prefix sum array initially filled with 0's
+	for(int i=1; i<=n; i++){
+		ll x; cin >> x;
+		dp[i] = dp[i-1] + x; // Compute the prefix sum at each element
+	}
+	ll max_subarray_sum = dp[0], min_prefix_sum = dp[0];
+	for(int i=1; i<=n; i++){
+		// Max subarray sum is the maximum difference between two prefix sums
+		max_subarray_sum = max(max_subarray_sum, dp[i] - min_prefix_sum);
+		min_prefix_sum = min(min_prefix_sum, dp[i]);
+	}
+	cout << max_subarray_sum << endl;
+}
+```
+
+</CPPSection>
+
+</LanguageSection>
+
+</Spoiler>
+
+<Warning>
+
 This problem has a solution known as [Kadane's Algorithm](https://en.wikipedia.org/wiki/Maximum_subarray_problem#Kadane's_algorithm). Please don't use that solution; try to solve it with prefix sums.
 
-<Spoiler title="Why are the two methods equivalent?">
+</Warning>
+
+<Spoiler title = "Alternative Solution: Kadane's Algorithm">
+
+Kadane's Algorithm finds the largest subarray sum using a greedy method. Check the video below for details:
+
+[Video explanation of the max subarray sum using Kadane's Algorithm](https://www.youtube.com/watch?v=jnoVtCKECmQ&t=611s).
```

Comment:
nick white...

Reply:
idk what are our opinions on putting nick white in guide LOL

Reply:
I mean he explains Kadane's fairly well lol... Also, not many ppl will look at Kadane's solution anyway lol..

Reply:
> nick white...

All that matters is that the approach is explained in a clear and logical way. Idt the channel really matters tbh. What matters is whether u could clearly learn, right??

Reply:
I don't really like the explanation tbh, it seems geared towards an interview and not towards CP in general, so it's really long...

Reply:
If that's the case, then should we remove the Kadane's algorithm sol?